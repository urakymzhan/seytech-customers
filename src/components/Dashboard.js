import React from 'react';

export default function Dashboard() {
  return (
    <div>
      <h1>WELCOME TO SEYTECH PORTAL</h1>
      <h3>Profile</h3>
      <h3>Calendar/Schedule</h3>
      <h3>News</h3>
      <h3>Job Fair</h3>
      <h3>Courses</h3>
      <ul>
        <li>Individual Course</li>
      </ul>
    </div>
  );
}
