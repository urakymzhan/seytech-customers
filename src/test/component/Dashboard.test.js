import React from 'react';
import Dashboard from '../../components/Dashboard';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('<Customers> component tests', () => {
  it('renders <Dashboard> component correctly', () => {
    shallow(<Dashboard />);
  });

  test('snapshot renders', () => {
    const component = renderer.create(<Dashboard />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot(); // https://jestjs.io/docs/en/snapshot-testing
  });

  it('Dashboard should have h1 text correctly', () => {
    let app = shallow(<Dashboard />);
    expect(app.find('h1').text()).toEqual('WELCOME TO SEYTECH PORTAL');
  });

  it('Dashboard should have h3 at 0 with Profile text', () => {
    let app = shallow(<Dashboard />);
    expect(app.find('h3').at(0).text()).toEqual('Profile');
  });
  it('Dashboard should have h3 at 1 with Calendar/Schedule text', () => {
    let app = shallow(<Dashboard />);
    expect(app.find('h3').at(1).text()).toEqual('Calendar/Schedule');
  });
});
