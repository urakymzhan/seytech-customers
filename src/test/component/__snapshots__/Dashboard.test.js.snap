// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Customers> component tests snapshot renders 1`] = `
<div>
  <h1>
    WELCOME TO SEYTECH PORTAL
  </h1>
  <h3>
    Profile
  </h3>
  <h3>
    Calendar/Schedule
  </h3>
  <h3>
    News
  </h3>
  <h3>
    Job Fair
  </h3>
  <h3>
    Courses
  </h3>
  <ul>
    <li>
      Individual Course
    </li>
  </ul>
</div>
`;
